components/compliance-service/reporting/relaxting/reports.go

```diff
@@ -645,14 +645,58 @@ func (backend *ES2Backend) GetControlListItems(ctx context.Context, filters map[
 	controlTermsAgg.SubAggregation("title",
 		elastic.NewTermsAggregation().Field("profiles.controls.title").Size(1))
 
-	controlTermsAgg.SubAggregation("skipped",
-		elastic.NewFilterAggregation().Filter(elastic.NewTermQuery("profiles.controls.status", "skipped")))
+	waivedQuery := elastic.NewTermsQuery("profiles.controls.waived_str", "yes", "yes_run")
+	passedFilter := elastic.NewFilterAggregation().Filter(elastic.NewBoolQuery().
+		Must(elastic.NewTermQuery("profiles.controls.status", "passed")).
+		MustNot(waivedQuery))
 
-	controlTermsAgg.SubAggregation("failed",
-		elastic.NewFilterAggregation().Filter(elastic.NewTermQuery("profiles.controls.status", "failed")))
+	failedFilter := elastic.NewFilterAggregation().Filter(elastic.NewBoolQuery().
+		Must(elastic.NewTermQuery("profiles.controls.status", "failed")).
+		MustNot(waivedQuery))
 
-	controlTermsAgg.SubAggregation("passed",
-		elastic.NewFilterAggregation().Filter(elastic.NewTermQuery("profiles.controls.status", "passed")))
+	skippedFilter := elastic.NewFilterAggregation().Filter(elastic.NewBoolQuery().
+		Must(elastic.NewTermQuery("profiles.controls.status", "skipped")).
+		MustNot(waivedQuery))
+
+	waivedFilter := elastic.NewFilterAggregation().Filter(waivedQuery)
+
+	controlTermsAgg.SubAggregation("skipped", skippedFilter)
+	controlTermsAgg.SubAggregation("failed", failedFilter)
+	controlTermsAgg.SubAggregation("passed", passedFilter)
+	controlTermsAgg.SubAggregation("waived", waivedFilter)
+
+	waivedStrAgg := elastic.NewTermsAggregation().Field("profiles.controls.waived_str").Size(4) //there are 4 different waived_str states
+	waivedButNotRunQuery := elastic.NewTermsQuery("profiles.controls.waived_str", "yes")
+	waivedAndRunQuery := elastic.NewTermsQuery("profiles.controls.waived_str", "yes_run")
+	waiverDataWaivedQuery := elastic.NewTermsQuery("profiles.controls.waived_str", "yes", "yes_run")
```

Comment:
This is the same variable as on line 648 `waivedQuery`. Could you just use the one variable?

Reply:
Good catch @lancewf, thanks. I have removed waiverDataWaivedQuery